buildSrc/src/main/java/org/opensearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -204,24 +206,27 @@ static void createBuildBwcTask(
         TaskProvider<Task> bwcTaskProvider
     ) {
         String bwcTaskName = buildBwcTaskName(projectName);
-        bwcSetupExtension.bwcTask(bwcTaskName, c -> {
-            c.getInputs().file(new File(project.getBuildDir(), "refspec"));
-            c.getOutputs().files(projectArtifact);
-            c.getOutputs().cacheIf("BWC distribution caching is disabled on 'master' branch", task -> {
-                String gitBranch = System.getenv("GIT_BRANCH");
-                return BuildParams.isCi() && (gitBranch == null || gitBranch.endsWith("master") == false);
-            });
-            c.args(projectPath.replace('/', ':') + ":assemble");
-            if (project.getGradle().getStartParameter().isBuildCacheEnabled()) {
-                c.args("--build-cache");
-            }
-            c.doLast(task -> {
-                if (projectArtifact.exists() == false) {
-                    throw new InvalidUserDataException(
-                        "Building " + bwcVersion.get() + " didn't generate expected file " + projectArtifact
-                    );
+        bwcSetupExtension.bwcTask(bwcTaskName, new Action<LoggedExec>() {
```

Comment:
Reason: Using Java lambdas is not supported as task inputs. Please refer to https://docs.gradle.org/7.3/userguide/validation_problems.html#implementation_unknown for more details about this problem.